addons/website_blog/static/src/js/contentshare.js

```diff
@@ -3,107 +3,132 @@
-            $.fn.share.defaults.shareable.on('mouseup', function () {
-                if ($(this).parents('body.editor_enable').length === 0) {
-                    self.popOver();
-                }
-            });
-            $.fn.share.defaults.shareable.on('mousedown', function () {
-                self.destroy();
-            });
-        },
-        getContent: function () {
-            var $popover_content = $('<div class="h4 m-0"/>');
-            if ($('.o_wblog_title, .o_wblog_post_content_field').hasClass('js_comment')) {
-                selected_text = this.getSelection('string');
-                var $btn_c = $('<a class="o_share_comment btn btn-link px-2" href="#"/>').append($('<i class="fa fa-lg fa-comment"/>'));
-                $popover_content.append($btn_c);
-            }
-            if ($('.o_wblog_title, .o_wblog_post_content_field').hasClass('js_tweet')) {
-                var tweet = '"%s" - %s';
-                var baseLength = tweet.replace(/%s/g, '').length;
-                // Shorten the selected text to match the tweet max length
-                // Note: all (non-localhost) urls in a tweet have 23 characters https://support.twitter.com/articles/78124
-                var selectedText = this.getSelection('string').substring(0, option.maxLength - baseLength - 23);
-
-                var text = window.btoa(encodeURIComponent(sprintf(tweet, selectedText, window.location.href)));
-                $popover_content.append(sprintf(
-                    "<a onclick=\"window.open('%s' + atob('%s'), '_%s','location=yes,height=570,width=520,scrollbars=yes,status=yes')\"><i class=\"ml4 mr4 fa fa-twitter fa-lg\"/></a>",
-                    option.shareLink, text, option.target));
-            }
-            return $popover_content;
-        },
-        commentEdition: function () {
-            $(".o_portal_chatter_composer_form textarea").val('"' + selected_text + '" ').focus();
-            const commentsEl = $('#o_wblog_post_comments')[0];
-            if (commentsEl) {
-                scrollTo(commentsEl).then(() => {
-                    window.location.hash = 'blog_post_comment_quote';
-                });
-            }
-        },
-        getSelection: function (share) {
-            if (window.getSelection) {
-                var selection = window.getSelection();
-                if (!selection || selection.rangeCount === 0) {
-                    return "";
-                }
-                if (share === 'string') {
-                    return String(selection.getRangeAt(0)).replace(/\s{2,}/g, ' ');
-                } else {
-                    return selection.getRangeAt(0);
-                }
-            } else if (document.selection) {
-                if (share === 'string') {
-                    return document.selection.createRange().text.replace(/\s{2,}/g, ' ');
-                } else {
-                    return document.selection.createRange();
-                }
-            }
```

Comment:
@robinlej Its just the diff of the optimization i did of getSelection method, it basically does the same thing:

previeously it was :
    
```js
        getSelection: function (share) {
            if (window.getSelection) {
                var selection = window.getSelection();
                if (!selection || selection.rangeCount === 0) {
                    return "";
                }
                if (share === 'string') {
                    return String(selection.getRangeAt(0)).replace(/\s{2,}/g, ' ');
                } else {
                    return selection.getRangeAt(0);
                }
            } else if (document.selection) {
                if (share === 'string') {
                    return document.selection.createRange().text.replace(/\s{2,}/g, ' ');
                } else {
                    return document.selection.createRange();
                }
            }
        },

                            NOW: 

    function getSelection(type) {
        const selection = window.getSelection();
        if (!selection || selection.rangeCount === 0) {
            return "";
        }
        if (type === "string") {
            return String(selection.getRangeAt(0)).replace(/\s{2,}/g, " ");
        } else {
            return selection.getRangeAt(0);
        }
    }
```

Reply:
I don't understand how `window.getSelection` could be `false` before, but in such a case that means now you don't have a `selection` and you're not doing the same as before. :raised_eyebrow: 
I'm not saying it's wrong, but I don't get how both codes are equivalent.

Reply:
@robinlej The first code handles both window.getSelection and document.selection (for older IE versions).
The second code uses only window.getSelection, ideal for modern browsers.
Both codes use window.getSelection to get the current text selection.
They check if the selection is empty or if rangeCount is zero, returning an empty string if so.
The second code is simpler for modern environments, assuming no legacy support is needed.
Since Internet Explorer is no longer used, the condition for `document.selection` isn't necessary.
I hope i am able to explain the changes i made there ? :sweat_smile:

Reply:
Oh, alright, I didn't know `document.selection` was for IE, that's the info I missed. Thanks :+1: